hoodie-client/src/main/java/com/uber/hoodie/io/HoodieInsertHandle.java

```diff
@@ -82,13 +84,19 @@ public boolean canWrite(HoodieRecord record) {
      */
     public void write(HoodieRecord record) {
         try {
-            IndexedRecord avroRecord = record.getData().getInsertValue(schema);
-            storageWriter.writeAvroWithMetadata(avroRecord, record);
-            status.markSuccess(record);
-            // update the new location of record, so we know where to find it next
-            record.setNewLocation(new HoodieRecordLocation(commitTime, status.getFileId()));
+            Optional<IndexedRecord> avroRecord = record.getData().getInsertValue(schema);
+
+            if(avroRecord.isPresent()) {
+                storageWriter.writeAvroWithMetadata(avroRecord.get(), record);
+                // update the new location of record, so we know where to find it next
+                record.setNewLocation(new HoodieRecordLocation(commitTime, status.getFileId()));
+                recordsWritten++;
+            } else {
+                recordsDeleted++;
```

Comment:
This will NPE here - https://github.com/uber/hoodie/blob/master/hoodie-client/src/main/java/com/uber/hoodie/index/HBaseIndex.java#L188. 

We should ideally make HoodieRecordLocation optional now that deletes mean newLocation is empty.